cmake: silence "LOAD segment with RWX permissions" warning

marckleinebudde opened the pull request:
Link: https://github.com/candle-usb/candleLight_fw/issues/106

marckleinebudde committed:
cmake: silence "LOAD segment with RWX permissions" warning

Link: https://github.com/candle-usb/candleLight_fw/issues/106

marckleinebudde commented:
@fenugrec my compiler doesn't support `--no-warn-rwx-segments` and `cmake` doesn't use the new option. Does it fix the warning with your compiler?

marckleinebudde commented:
Debian stable ("bullseye") uses cmake-3.18, Debian oldstable ("buster") is still on cmake-3.13, but "buster-backports" has cmake-3.18, too.

fenugrec commented:
I think bumping required to 3.18 is reasonable; distros with really old cmake have a simple workaround of just reverting this commit.

marckleinebudde commented:
Thanks for testing! Fixed.

fenugrec commented:
> Fixed.

Thanks, youǜve fixed the passing-wrong-arg problem, but since the test compilation fails, cmake still thinks the flag is "not supported"

marckleinebudde commented:
Just noticed, too. But why does it say `cached: true`.

fenugrec commented:
> But why does it say `cached: true`.

AFAIK because https://cmake.org/cmake/help/latest/module/CheckLinkerFlag.html

>  Stores the result in an internal cache entry named <var>.

marckleinebudde committed:
workflows: update CI to compile on various Debian based distributions

Use containers to build on various Debian based distributions:

- Ubuntu: 20.04
- Ubuntu: 22.04
- Ubuntu: 24.04
- Ubuntu: rolling
- Debian: oldstable
- Debian: stable
- Debian: testing
- Debian: unstable

marckleinebudde committed:
editorconfig: apply indent with 2 spaces for yml files

This is the default for yml files for the github workflows.

marckleinebudde committed:
workflows: pr: update actions/checkout to v4
